MobileWallet/Screens/AppEntry/Splash/SplashViewController.swift

```diff
@@ -323,6 +323,9 @@ class SplashViewController: UIViewController, UITextViewDelegate {
 
     private func navigateToHome() {
         if walletExistsInitially {
+            //Calling this here in case they did not succesfully register the token in the onboarding
+            NotificationManager.shared.retryIfTokenNotRegistered()
```

Comment:
You can move the checks in to `requestAuthorization` & get rid of `retryIfTokenNotRegistered` & call `requestAuthorization` directly - delegate all the checks to `requestAuthorization`.